### PR TITLE
prov/efa: Decouple device from fabric id

### PR DESCRIPTION
**Pull request by wckzhang**
Previously we used our device GID as fabric id, which
is not an appropriate coupling. Since our GID does not
contain information to indicate connectivity, we use a
static/constant fabric id instead.

Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

**Comment by wckzhang** [comment]
fi_info -p efa output:
```
provider: efa
    fabric: efa
    domain: efa_0-rdm
    version: 114.0
    type: FI_EP_RDM
    protocol: FI_PROTO_EFA
provider: efa
    fabric: efa
    domain: efa_0-dgrm
    version: 114.0
    type: FI_EP_DGRAM
    protocol: FI_PROTO_EFA
```

File: prov/efa/src/efa_fabric.c [hunk 1]
```diff
@@ -710,14 +710,15 @@ static int efa_alloc_info(struct efa_device *device, struct fi_info **info,
 	if (ret)
 		goto err_free_info;
 
-	name_len = strlen(EFA_FABRIC_PREFIX) + INET6_ADDRSTRLEN;
+	name_len = 3;
```
Comment (wzamazon) [inline prov/efa/src/efa_fabric.c:713]: Instead of using a number, Maybe define a constant `fabbric_name` and apply `strlen` on it?

**Review by wzamazon — COMMENTED** [review]
Can you add a test case in pytest/efa so the behavior will not regress?

File: prov/efa/src/efa_fabric.c (hunk 1 above)
Comment (wckzhang) [reply prov/efa/src/efa_fabric.c:713]: Sure

**Comment by wckzhang** [comment]
I'll add the pytest/efa test

File: fabtests/prov/efa/src/efa_info_test.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	hints = fi_allocinfo();
+	if (!hints)
+		return EXIT_FAILURE;
+	hints->fabric_attr->prov_name=strdup("efa");
+	ret = ft_init();
+	if (ret) {
+		FT_PRINTERR("ft_init", -ret);
+		goto out;;
+	}
+	ret = ft_getinfo(hints, &fi);
```
Comment (wzamazon) [inline fabtests/prov/efa/src/efa_info_test.c:50]: Where is `fi` defined?

File: fabtests/prov/efa/src/efa_info_test.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	if (ret) {
+		FT_PRINTERR("ft_getinfo", -ret);
+		goto out;
+	}
+	while (NULL != fi) {
+		if (0 != strcmp(fi->fabric_attr->name, "efa")) {
+			return EXIT_FAILURE;
```
Comment (wzamazon) [inline fabtests/prov/efa/src/efa_info_test.c:57]: Please free `hints` and `fi` upon exit.

**Review by wzamazon — COMMENTED** [review]
1st commit looks good.

I left some comments on 2nd commit.

File: fabtests/prov/efa/src/efa_info_test.c (hunk 2 above)
Comment (wckzhang) [reply fabtests/prov/efa/src/efa_info_test.c:50]: In shared.h, it's one of the fabtest globals

Comment (wzamazon) [reply fabtests/prov/efa/src/efa_info_test.c:50]: I would prefer to not use global variable if possible.

**Review by wzamazon — APPROVED** [review]

**Pull request merged**